servicetalk-grpc-api/src/main/java/io/servicetalk/grpc/api/GrpcRouter.java

```diff
@@ -214,19 +256,37 @@ RouteProviders drainRoutes() {
                 final StreamingRoute<Req, Resp> route, final Class<Req> requestClass,
                 final Class<Resp> responseClass, final GrpcSerializationProvider serializationProvider) {
             streamingRoutes.put(path, new RouteProvider(executionContext -> {
-                StreamingHttpService service = (ctx, request, responseFactory) -> {
-                    try {
-                        final GrpcServiceContext serviceContext = new DefaultGrpcServiceContext(request.path(), ctx);
-                        final HttpDeserializer<Req> deserializer =
-                                serializationProvider.deserializerFor(readGrpcMessageEncoding(request), requestClass);
-                        final Publisher<Resp> response = route.handle(serviceContext, request.payloadBody(deserializer))
-                                .map(GrpcUtils::uncheckedCast);
-                        return succeeded(newResponse(responseFactory, response,
-                                serializationProvider.serializerFor(serviceContext, responseClass),
-                                ctx.executionContext().bufferAllocator()));
-                    } catch (Throwable t) {
-                        return succeeded(newErrorResponse(responseFactory, t,
-                                ctx.executionContext().bufferAllocator()));
+                StreamingHttpService service = new StreamingHttpService() {
+                    @Override
+                    public Single<StreamingHttpResponse> handle(final HttpServiceContext ctx,
+                                                                final StreamingHttpRequest request,
+                                                                final StreamingHttpResponseFactory responseFactory) {
+                        try {
+                            final GrpcServiceContext serviceContext =
+                                    new DefaultGrpcServiceContext(request.path(), ctx);
+                            final HttpDeserializer<Req> deserializer =
+                                    serializationProvider.deserializerFor(readGrpcMessageEncoding(request),
+                                            requestClass);
+                            final Publisher<Resp> response = route.handle(serviceContext,
+                                    request.payloadBody(deserializer))
+                                    .map(GrpcUtils::uncheckedCast);
```

Comment:
while you are here: `uncheckedCast` shouldn't be necessary (`Class<Resp>` should be used to give type safety) ... consider removal.